src/mds/MDCache.cc

```diff
@@ -12035,6 +12035,28 @@ void MDCache::show_subtrees(int dbl)
     }
   }
 
+  if (subtrees_seen.size() > SUBTREES_COUNT_THRESHOLD && depth >
+      SUBTREES_DEPTH_THRESHOLD) {
+    dout(15) << "the number of subtrees and depth is beyond the threshold."
+	     << dendl;
+    dout(15) << "number of subtrees = " << subtrees_seen.size() << ", max "
+		"depth among all the subtrees = " << depth << dendl;
+    dout(15) << "SUBTREES_COUNT_THRESHOLD = " << SUBTREES_COUNT_THRESHOLD
+	     << ", SUBTREES_DEPTH_THRESHOLD = " << SUBTREES_DEPTH_THRESHOLD
+	     << dendl;
+  } else if (subtrees_seen.size() > SUBTREES_COUNT_THRESHOLD) {
+    dout(15) << "the number of subtrees is beyond the threshold." << dendl;
+    dout(15) << "number of subtrees = " << subtrees_seen.size() << dendl;
+    dout(15) << "SUBTREES_COUNT_THRESHOLD = " << SUBTREES_COUNT_THRESHOLD
+	     << dendl;
+  } else if (depth > SUBTREES_DEPTH_THRESHOLD) {
+    dout(15) << "the depth of subtrees is beyond the threshold." << dendl;
+    dout(15) << "max depth = " << depth << dendl;
+    dout(15) << "SUBTREES_DEPTH_THRESHOLD = " << SUBTREES_DEPTH_THRESHOLD
+	     << dendl;
+  }
+  //if (current_logging_level < 25)
```

Comment:
@batrick How do I get the current logging level? I tried `g_conf().get_val<Option::size_t>("debug_mds")` but it didn't work for me.

Reply:
See: https://github.com/ceph/ceph/blob/2c6a785410311041a86abdad235ebc3d1830813d/src/mds/Migrator.cc#L622

Reply:
`should_gather` returns a boolean indicating whether the log message would be generated. i.e. 

`g_conf()->subsys.should_gather<ceph_subsys_mds, 5>()`

returns true if

`dout(5) << foo << dendl`

would print something.

Or am I misunderstanding you?

Reply:
You are right. Initially, I felt that default value of `debug mds` was being returned by `g_conf()->subsys.should_gather<ceph_subsys_mds, 5>()` (since it always returned, as I report in my previous comment, and the default value for `debug mds` is also 1). So, I was looking for another function that would return the current value.